src/wagtail_ai/wagtail_hooks.py

```diff
@@ -83,18 +83,17 @@ def get_prompts():
 
 @hooks.register("insert_editor_js")  # type: ignore
 def ai_editor_js():
-    prompt_json = json.dumps(get_prompts(), cls=DjangoJSONEncoder)
+    prompt_json = get_prompts()
     process_url = reverse("wagtail_ai:process")
 
-    return mark_safe(
-        f"""
-        <script>
-            window.WAGTAIL_AI_PROCESS_URL = "{process_url}";
-            window.WAGTAIL_AI_PROMPTS = {prompt_json};
-        </script>
-        """
+    wagtail_ai_config = json_script(
+        {"wagtail_ai_prompts": prompt_json, "wagtail_ai_process_url": process_url},
+        "wagtail-ai-config",
+        DjangoJSONEncoder,
```

Comment:
I think this should be a keyword argument.
```suggestion
        encoder=DjangoJSONEncoder,
```

Reply:
Changing this to a keyword argument doesn't fix the type issue that's been raised by Pyright.

I think it's related to [django-types](https://github.com/sbdchd/django-types/blob/d44899f411df6036610b81cc06cddccd8f1e17cd/django-stubs/utils/html.pyi#L17) only listing two arguments for json_script, whereas it should accept 3 ([django repo link](https://github.com/django/django/blob/main/django/utils/html.py#L85)).

Reply:
Good catch. As the default encoder for `json_script` is `DjangoJSONEncoder` we can probably just drop that argument here?

Reply:
Thanks Tom, I missed that while I was trying to find out why the type check didn't work.

Reply:
Resolved in 81985f16ee6886d79a95732f928e42a9bb1a6bbc.